src/main/java/net/whydah/service/auth/UserAuthenticationResource.java

```diff
@@ -0,0 +1,157 @@
+    private static final Logger log = LoggerFactory.getLogger(UserAuthenticationResource.class);
+
+    private final SPAApplicationRepository spaApplicationRepository;
+
+    @Autowired
+    public UserAuthenticationResource(CredentialStore credentialStore, SPAApplicationRepository spaApplicationRepository) {
+        super(credentialStore);
+        this.spaApplicationRepository = spaApplicationRepository;
+    }
+
+    @POST
+    @Path("/{secret}/authenticate_user/")
+    public Response authenticateUser(@Context HttpServletRequest httpServletRequest,
+                                     @Context HttpServletResponse httpServletResponse,
+                                     @Context HttpHeaders headers,
+                                     @PathParam("secret") String secret,
+                                     @FormParam("username") String username,
+                                     @FormParam("password") String password) {
+        log.info("Invoked authenticateUser with secret: {} and headers: {}", secret, headers.getRequestHeaders());
+
+        if (username == null || password == null) {
+            log.warn("Unable to find the user credential, returning 403");
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+
+        // 1. lookup secret in secret-application session map
+        ApplicationToken applicationToken = spaApplicationRepository.getApplicationTokenBySecret(secret);
+        // 2. execute Whydah API request using the found application
+        if (applicationToken == null) {
+            log.warn("Unable to locate applicationsession from secret, returning 403");
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        Application application = credentialStore.findApplication(applicationToken.getApplicationName());
+
+        UserCredential userCredential = new UserCredential(username, password);
+
+        String ticket = UUID.randomUUID().toString();
+        UserToken userToken = UserTokenMapper.fromUserTokenXml(new CommandLogonUserByUserCredential(
+                URI.create(credentialStore.getWas().getSTS()), applicationToken.getApplicationTokenId(),
+                ApplicationTokenMapper.toXML(applicationToken), userCredential, ticket).execute());
+        if (userToken == null) {
+            // Most likely timeout in application sesssion, lets create a new here..
+            ApplicationCredential appCredential = new ApplicationCredential(application.getId(), application.getName(),
+                    application.getSecurity().getSecret());
+            String myAppTokenXml = new CommandLogonApplication(URI.create(credentialStore.getWas().getSTS()), appCredential).execute();
+            applicationToken = ApplicationTokenMapper.fromXml(myAppTokenXml);
+            spaApplicationRepository.add(secret, applicationToken);
+            userToken = UserTokenMapper.fromUserTokenXml(new CommandLogonUserByUserCredential(
```

Comment:
As mentioned in https://github.com/Cantara/Whydah-SPAProxyService/pull/5/files#r214604011 - this is the same functionality right?